### ISSUE TITLE
Features request - cube, rollup, grouping sets, grouping_id; official .net driver

### ISSUE THREAD

**Issue by brasio** [duckdb/duckdb#2156]
Hi, is there any plan to see these analytic features implemented? I'm actually using HSQL db and it seems this is the only embedded db having these functions.
I have searched for a .net official driver too, but without success. Is it planned in the future?
Thanks. Best Regards.

**Comment by hannesmuehleisen** [comment]
There is a lot here, indeed we should eventually support the 

- [ ] `CUBE`
- [ ] `ROLLUP`
- [ ] Grouping Sets

I'm not familiar with "grouping_id", did you refer to https://docs.microsoft.com/en-us/sql/t-sql/functions/grouping-id-transact-sql ?

And regarding .NET, would the new ODBC driver not be an option there?

**Comment by brasio** [comment]
Yes, grouping_id is the function you mean: it's a standard sql function (I don't remember which ansi version), Oracle, Postgres support it too. Performance with these functions are really optimized in these db: orders of magnitude higher than the simple sql "group by" statements that they replace. 

Regarding .Net, it would be useful a dll which works with the c++ official dll (similar to sqlite .Net), without installing any driver. There is an unofficial project on GitHub (Giorgi/DuckDB.NET), but it is in very early stage.

**Comment by Giorgi** [comment]
Hello @brasio 

I'm the author of the DuckDB .Net library. Even though the library isn't fully implemented, it is ADO.NET compliant and supports all basic operations. You can create tables, and run Select/Update/Insert commands. Is there a specific functionality that you need that is missing from the library?

Also, if you want to support the project why don't you contribute to it or show your appreciation by [supporting the project](https://ko-fi.com/U6U81LHU8)?
